core/modules/node/node.types.inc

```diff
@@ -683,11 +727,26 @@ function node_type_delete_confirm($form, &$form_state, $type) {
  * @see node_type_delete_confirm()
  */
 function node_type_delete_confirm_submit($form, &$form_state) {
-  $type_name = $form_state['values']['type'];
-  node_type_delete($type_name);
+  $type = node_type_load($form['type']['#value']);
+
+  $nodes = db_select('node', 'n')
```

Comment:
Instead of db_select() you could use db_query() here, which is faster.

See the explanation [here](https://api.backdropcms.org/api/backdrop/core%21includes%21database%21database.inc/function/db_query/1)